level/level.c

```diff
@@ -186,6 +186,15 @@ level* make_level(long int map_seed) {
     //TODO have make_map() return the starting coords for the player based on root room
     make_map(lvl);
 
+    if (reveal_map) {
```

Comment:
Make this it's own function and call it from the main rather than passing `reveal_map` into `make_level`. That gives us more flexibility to manipulate the FOW and it removes unnecessary complexity from this over burdened part of the code.